base/show.jl

```diff
@@ -597,6 +597,62 @@ function make_typealias(@nospecialize(x::Type))
     end
 end
 
+function show_can_elide(p::TypeVar, wheres::Vector, elide::Int, env::SimpleVector, skip::Int)
+    elide == 0 && return false
+    wheres[elide] === p || return false
+    for i = (elide + 1):length(wheres)
+        v = wheres[i]::TypeVar
+        has_typevar(v.lb, p) && return false
+        has_typevar(v.ub, p) && return false
+    end
+    for i = 1:length(env)
+        i == skip && continue
+        has_typevar(env[i], p) && return false
+    end
+    return true
+end
+
+function show_typeparams(io::IO, env::SimpleVector, orig::SimpleVector, wheres::Vector)
+    n = length(env)
+    elide = length(wheres)
+    for i = n:-1:1
+        p = env[i]
+        if p isa TypeVar
```

Comment:
maybe reduce nesting: `p isa TypeVar || continue`?

Reply:
there is an else clause, so this is the right structure

Reply:
oh, I see this is the other loop. The reason is still the same though (keeping the general structure of the two loops in common)